vault/expiration.go

```diff
@@ -2275,6 +2300,44 @@ func (m *ExpirationManager) walkLeases(walkFn leaseWalkFunction) error {
 	return nil
 }
 
+// must be called with m.pendingLock held
+func (m *ExpirationManager) removeFromPending(ctx context.Context, leaseID string) {
+	if info, ok := m.pending.Load(leaseID); ok {
+		pending := info.(pendingInfo)
+		pending.timer.Stop()
+		m.pending.Delete(leaseID)
+		m.leaseCount--
+		// Log but do not fail; unit tests (and maybe Tidy on production systems)
+		if err := m.core.quotasHandleLeases(ctx, quotas.LeaseActionDeleted, []string{leaseID}); err != nil {
+			m.logger.Error("failed to update quota on revocation", "error", err)
+		}
+	}
+}
+
+func (m *ExpirationManager) markLeaseAsZombie(ctx context.Context, le *leaseEntry, err error) {
+	m.pendingLock.Lock()
+	defer m.pendingLock.Unlock()
+	if le.isZombie() {
+		m.logger.Info("attempted to re-mark lease as zombie", "original_error", le.RevokeErr, "new_error", err.Error())
+		return
+	}
+
+	errStr := err.Error()
+	if len(errStr) == 0 {
+		errStr = "no error message given"
+	}
+	if len(errStr) > maxZombieErrorLength {
```

Comment:
Could we 'intern' these strings, e.g dedupe them in a separate data structure?  Or is that overkill and we're confident in the memory usage being limited by character length without losing info?

Reply:
hmm, i'm not 100% sure what you're asking. could you phrase it differently or expand on it?

the main concern here is we don't want to accidentally over-consume storage (and memory) with many zombies that might have long (i.e. wrapped) errors